Explorer/Assets/Scripts/Global/Dynamic/IRealmController.cs

```diff
@@ -1,20 +1,19 @@
 using CommunicationData.URLHelpers;
 using Cysharp.Threading.Tasks;
+using DCL.AsyncLoadReporting;
 using System.Threading;
+using UnityEngine;
 
 namespace Global.Dynamic
 {
     public interface IRealmController
     {
-        /// <summary>
-        ///     Setup the GlobalWorld
-        /// </summary>
-        void SetupWorld(GlobalWorld world);
+        GlobalWorld GlobalWorld { set; }
 
         /// <summary>
         ///     Unload the current realm and load the new one
         /// </summary>
-        UniTask SetRealmAsync(URLDomain realm, CancellationToken ct);
+        UniTask SetRealmAsync(URLDomain realm, Vector2Int playerStartPosition, AsyncLoadProcessReport? loadReport, CancellationToken ct);
```

Comment:
Let's assume `AsyncLoadProcessReport` is never null. Then you won't need to check it every time: it looks dirty